pkg/datagatherer/versionchecker/versionchecker_test.go

```diff
@@ -0,0 +1,357 @@
+// to load in for the DataGatherer
+func createDgHostConfigWithServer(server string) (*os.File, error) {
+	tmpfile, err := ioutil.TempFile("", "example")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a tmpfile for host")
+	}
+
+	if _, err := tmpfile.Write([]byte(server)); err != nil {
+		return nil, fmt.Errorf("failed to write to tmp host file")
+	}
+	if err := tmpfile.Close(); err != nil {
+		return nil, fmt.Errorf("failed to close tmp host file after writing")
+	}
+
+	return tmpfile, nil
+}
+
+// creates a kubeconfig file on disk with a reference to the local server
+// allowing requests to be mocked
+func createKubeConfigWithServer(server string) (*os.File, error) {
+	content := fmt.Sprintf(`
+apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    server: %s
+  name: example
+contexts:
+- context:
+    cluster: example
+    namespace: default
+    user: test
+  name: test
+current-context: test
+users:
+- name: test
+  user:
+    username: test
+    password: test`, server)
+	tmpfile, err := ioutil.TempFile("", "example")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a tmpfile for kubeconfig")
+	}
+
+	if _, err := tmpfile.Write([]byte(content)); err != nil {
+		return nil, fmt.Errorf("failed to write to tmp kubeconfig file")
+	}
+	if err := tmpfile.Close(); err != nil {
+		return nil, fmt.Errorf("failed to close tmp kubeconfig file after writing")
+	}
+
+	return tmpfile, nil
+}
+
+// create a local test server to respond to k8s and registry api requests from
+// the DataGatherer during operation. The dg is configured to use this local
+// address to get data from k8s and registries.
+func createLocalTestServer(t *testing.T) *httptest.Server {
+	var localServer *httptest.Server
+	localServer = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var responseContent []byte
+
+		if r.URL.Path == "/api/v1/pods" {
+			// the responses from the server are self referential and the host is
+			// needed to generate responses
+			parsedURL, err := url.Parse(localServer.URL)
+			if err != nil {
+				t.Fatalf("failed to parse test server url %s", localServer.URL)
+			}
+
+			tmpl, err := template.ParseFiles("fixtures/pods.json.tmpl")
+			if err != nil {
+				t.Fatalf("failed to load template files: %s", err)
+			}
+
+			// generate a response that contains pods pointing to this server
+			// as the registry
+			var response bytes.Buffer
+			err = tmpl.Execute(&response, struct{ URL *string }{&parsedURL.Host})
+			if err != nil {
+				t.Fatalf("failed to exe template: %s", err)
+			}
+			responseContent = response.Bytes()
+		} else if r.URL.Path == "/v2/jetstack/example/tags/list" {
+			file, err := os.Open("fixtures/tags.json")
+			if err != nil {
+				t.Fatalf("failed to open tags fixture: %s", err)
+			}
+			defer file.Close()
+
+			responseContent, err = ioutil.ReadAll(file)
+			if err != nil {
+				t.Fatalf("failed to read tags fixture: %s", err)
+			}
```

Comment:
ioutil.ReadFile

Reply:
Fixed in 87384d2b9bee22da0a1c2ed3d5868fc1074eafe2